### ISSUE TITLE
Summarize duplicate (label) changes in multiple files

### ISSUE THREAD

**Issue by rouke-broersma** [infro-io/issues#4]
Labels often reference the app version or chart version number. This causes a diff on practically every file in an application. It would be great if there was an option to somehow summarize this change. For example:

# Global changes

## Labels (25 files changed)
```
-    app.kubernetes.io/version: 56.6.2
-    chart: kube-prometheus-stack-56.6.2
+    app.kubernetes.io/version: 56.8.0
+    chart: kube-prometheus-stack-56.8.0
```

See https://github.com/broersma-forslund/homelab/pull/153#issuecomment-1952212701 for why this is needed.

I realize this might mean processing the diff as received from argocd which might not be in scope at this time. This might even be something that should be an option in argocd instead of in infro, because it's disturbing in the argocd UI as well.

**Comment by colethienes** [comment]
Yeah I guess makes sense with generated code like this. How would the output look here? Maybe show the diff and then associated changed manifests?

**Comment by rouke-broersma** [comment]
> Yeah I guess makes sense with generated code like this. How would the output look here? Maybe show the diff and then associated changed manifests?

I would like the example I gave in the issue body. It would need to be grouped somehow, perhaps by parent key for example (labels). 

If a file has other changes, that file should be shown as usual. The global changes in that file could be collapsed by default so only the changes that require reviewing are visible by default.

Files that have no other changes except the global changes can be fully collapsed by default.

**Comment by colethienes** [comment]
In your code comment, I think Infro will need access to read the source code in order to tell that the changed "things" are labels. Without access to the source code, it may be possible to do something like:

# Global changes

## ConfigMap (25 files changed)
```
-    app.kubernetes.io/version: 56.6.2
-    chart: kube-prometheus-stack-56.6.2
+    app.kubernetes.io/version: 56.8.0
+    chart: kube-prometheus-stack-56.8.0
```
<img src="https://argo-cd.readthedocs.io/en/stable/assets/favicon.png" width="20"/> **homelab > app-foo > prometheus/kube-prometheus-stack-alertmanager-overview**
<img src="https://argo-cd.readthedocs.io/en/stable/assets/favicon.png" width="20"/> **homelab > app-bar > prometheus/kube-prometheus-stack-apiserver**
<img src="https://argo-cd.readthedocs.io/en/stable/assets/favicon.png" width="20"/> **homelab > app-baz > prometheus/kube-prometheus-stack-cluster-total**

# Other changes
<img src="https://argo-cd.readthedocs.io/en/stable/assets/favicon.png" width="20"/> **homelab > app-foo**
```
bla bla bla
```

**Comment by rouke-broersma** [comment]
Infro already has access to the generated code through argocd does it not? But yes I see the issue, it is not available in the diff in all cases because the key might not be available in the diff view.


I'm not sure what this shows exactly:

 homelab > app-foo > prometheus/kube-prometheus-stack-alertmanager-overview
 homelab > app-bar > prometheus/kube-prometheus-stack-apiserver
 homelab > app-baz > prometheus/kube-prometheus-stack-cluster-total

Is the first level multiple apps, that also contain the same changes? In that case that's not exactly what I meant. I think 'global' changes should only ever be global for an app. Never across apps. The likelihood that they are in fact the same related change is much less across different apps.

I also don't think grouping by Kind  would work because the changed files are likely to be many different Kinds.

**Comment by colethienes** [comment]
Update - I've added all of the formatting [logic](https://github.com/infro-io/infro-core/blob/main/internal/github/format.go) to the open source repo, so I'll be thinking about how to do there.
